umbral/signing.py

```diff
@@ -21,7 +20,7 @@ class Signature:
     between (r, s) and DER formatting.
     """
 
-    def __init__(self, r: int, s: int):
+    def __init__(self, r: CurveBN, s: CurveBN):
```

Comment:
If we're going to use `CurveBN` here, this should be in a separate PR. I'm not quite sure what the best options here are yet and I want to think about it more.

Reply:
Yep, agreed.